common/swagger/v2/swagger.yaml

```diff
@@ -0,0 +1,71 @@
+openapi: '3.0.0'
+info:
+  description: >-
+    This is an extension of the REST API for scoring from H2O. This API is
+    intended to be used for scoring within H2O.ai MLOps.
+  version: 2.0.0
+  title: Scoring API - v2
+  contact:
+    email: support@h2o.ai
+  license:
+    name: License
+    url: 'http://www.h2o.ai'
+servers:
+  - url: '{protocol}://127.0.0.1:{port}/'
+    variables:
+      protocol:
+        enum:
+          - http
+          - https
+        default: http
+      port:
+        default: '8080'
+paths:
+  /model/score/media:
+    post:
+      tags:
+        - scoring
+      summary: Score model with provided media files
+      description: Computes score of provided data making use of provided media files
+      operationId: getScoreMedia
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                payload:
+                  $ref: '#/components/schemas/ScoreMediaRequest'
+                files:
+                  type: array
+                  items:
+                    type: string
+                    format: binary
+              required:
+                - payload
+                - files
+      responses:
+        '200':
+          description: Successful scoring operation
+          content:
+            application/json:
+              schema:
+                $ref: '../v1/swagger.yaml#/definitions/ScoreResponse'
+        '501':
+          description: Implementation not supported
+        '400':
+          description: Invalid payload
+components:
+  schemas:
+    ScoreMediaRequest:
+      allOf:
+        - $ref: '../v1/swagger.yaml#/definitions/ScoreRequest'
```

Comment:
So as to not be tied to the evolution of the non-media v1 spec, could this and the response have its own explicit definition?

Reply:
tbh I debated this as well... in fact, I even considered having a single request spec that both apis inherit from, because in reality it makes much more sense to have the requests and responses be unified and evolve independently of either api spec. Like each api should be dependent on the requests and responses themselves and while the apis can evolve they would be constrained by the requests/responses such that there is some level of consistency.